src/RewardsStreamerMP.sol

```diff
@@ -99,6 +100,60 @@ contract RewardsStreamerMP is IStakeManager, TrustedCodehashAccess, ReentrancyGu
         account.lastMPUpdateTime = block.timestamp;
     }
 
+    function lock(uint256 lockPeriod) external onlyTrustedCodehash nonReentrant {
+        if (lockPeriod == 0) {
+            revert StakingManager__InvalidLockingPeriod();
+        }
+
+        Account storage account = accounts[msg.sender];
+        if (account.stakedBalance == 0) {
+            revert StakingManager__InsufficientBalance();
+        }
+
+        if (account.lockUntil > 0) {
+            revert StakingManager__AlreadyLocked();
+        }
```

Comment:
We need to decide on whether we want this. See comment below.

Reply:
I think I prefer to have it, it make it simpler